### PR TITLE
Add KEP for Cron Horizontal Pod Autoscaler(CronHPA)

### PR DESCRIPTION
**Pull request by hex108**
I have implemented a CronHPA which enables us to scale workloads(those support scale subresource) periodically using crontab scheme.

## Use cases
A lot of game player will play games from Friday evening to Sunday evening. It will be better to provide a better experience for players if game servers(pods) could be scaled to a lager size at Friday evening and scaled to normal size at Sunday evening. And that's what game server admins do every week.

Some other customers also do similar things because their products' usage also have peaks and valleys periodically. CronHPA is for their requirement.

## CronHPA example:
```
---
apiVersion: gs.io/v1alpha1
kind: CronHPA
metadata:
  name: example-cron-hpa
spec:
  scaleTargetRef:
    apiVersion: extensions/v1beta1
    kind: Deployment
    name: demo-deployment
  crons:
    - schedule: "0 23 * * 5"  // Set replicas to 60 every Friday 23:00
      targetReplicas: 60
    - schedule: "0 23 * * 7"  // Set replicas to 30 every Sunday 23:00
      targetReplicas: 30

---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: demo-deployment
  labels:
    app: nginx
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx
        ports:
        - containerPort: 80
```

BTW: The original proposal was sent at https://github.com/kubernetes/autoscaler/issues/1439. Add a kep for it.

### CODE REVIEW

**Comment by hex108** [comment]
/sig autoscaling
cc @DirectXMan12

**Comment by justaugustus** [comment]
/ok-to-test

For review:
/assign @DirectXMan12

File: keps/sig-autoscaling/0032-20181127-cron-hpa.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+### Non-Goals
+
+* CronHPA does not support workloads that does not support `scale` subresource now.
+
+## Proposal
+
+At first, we'd like to use [CronJob](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) to implement this feature. Howerver `CronJob` only supports one `schedule` and users must specify an image and commands, it will be a little inappropriate for above use cases.
+
+We proposal a new CRD `CronHPA` which is inspried by CronJob(thanks a lot!) to meet users' requirement. Users could specify as many `schedule`  as they want.
```
Comment (justaugustus) [inline keps/sig-autoscaling/0032-20181127-cron-hpa.md]:
```suggestion
We proposal a new CRD `CronHPA` which is inspired by `CronJob` (thanks a lot!) to meet users' requirement. Users could specify as many `schedule`  as they want.
```

Comment (hex108) [reply keps/sig-autoscaling/0032-20181127-cron-hpa.md]: Thanks! Addressed.

**Comment by justaugustus** [comment]
KEPs [have moved](https://github.com/kubernetes/enhancements/pull/638) to [k/enhancements](https://github.com/kubernetes/enhancements). 
This PR will be closed and any additional changes to this KEP should be submitted to k/enhancements.
For more details on this change, review [this thread](https://groups.google.com/d/msg/kubernetes-dev/LIkZoIqCT20/t515CV2mAgAJ).

Any questions regarding this move should be directed to that thread and not asked on GitHub.
/close

**Pull request closed without merging**